src/table/utils/find-visible-rows.ts

```diff
@@ -36,6 +38,52 @@ const findVisibleRows = (rootElement: HTMLElement, totalsPosition: TotalsPositio
   const visibleRowStartIndex = findStartIndex(dataRows, yMin, yMax);
   const visibleRowEndIndex = findEndIndex(dataRows, yMin, yMax);
   return { visibleRowStartIndex, visibleRowEndIndex };
+}
+
+const getFristCellOfRow = (rowIndex: number, cells?: NodeListOf<Element>) => {
+  if (!cells) return undefined;
+  for (let i = 0; i < cells.length; i++) {
+    const strIdx = cells[i].getAttribute('rowindex');
+    if (strIdx && parseInt(strIdx, 10) === rowIndex) return cells[i];
+  }
+  return undefined;
+};
+
+const shouldCellBeIn = (cellRect: DOMRect, min: number, max: number) => {
+  if (!cellRect) return false;
+  const center = cellRect.y + cellRect.height / 2;
+  return center >= min && center <= max;
 };
 
-export default findVisibleRows;
+export function findVirtualizedVisibleRows(rootElement: HTMLElement, viewService: ViewService) {
+  const tableBody = rootElement.querySelector('.sn-table-body');
+  const bodyRect = tableBody?.getBoundingClientRect();
+  const cells = tableBody?.querySelectorAll('.sn-table-body .sn-table-cell');
+  if (!cells || !bodyRect) return {};
+  const { scrollTopRatio = 0, visibleTop = 0, visibleHeight = 0, page = 0, rowsPerPage = 0 } = viewService;
+  const offset = page * rowsPerPage;
+  const visibleTopInPage = visibleTop - offset;
+  const visibleBottomInPage = visibleTopInPage + visibleHeight - 1;
+  const r0c0Cell = getFristCellOfRow(visibleTopInPage, cells);
+  const r0c0CellRect = r0c0Cell?.getBoundingClientRect();
```

Comment:
these variable names are a bit cryptic. This is the top left cell right? Then it could be called `topLeftCell`

Reply:
Good point! Updated!